python/versionChecks.py

```diff
@@ -0,0 +1,92 @@
+import functools
+
+from vector.enums import ENVIRONMENT_VERSION_TYPE_T
+
+
+@functools.cache
+def vpythonHasCodedTestSupport():
+    """
+    Determine if the current version of vpython supports coded tests (result is
+    cached to avoid duplicate computation).
+
+    This function performs one check:
+
+        * Does it have the `Parser` package?
+    """
+
+    retVal = False
+    try:
+        from vector.lib.coded_tests import Parser
+
+        retVal = True
+    except ImportError:
+        pass
+    return retVal
+
+
+@functools.cache
+def vpythonHasCodedMockSupport():
+    """
+    Determine if the current version of vpython supports coded mocks (result is
+    cached to avoid duplicate computation).
+
+    This function performs three checks:
+
+        * Does it have the `mock_helper` package?
+
+        * Is `mock_helper` at the right version?
+
+        * Does it have environemnt revision REVISION_2024_CODED_MOCK_DATA?
+
+    NOTE: we don't need to check if an environment supports coded mocks, only a
+    version of _VectorCAST_.
+
+    If an environment doesn't have coded tests enabled (or it was originally
+    build with a version of VectorCAST that did not support coded tests/coded
+    mocks) and we open it up with a version of VectorCAST with `mock_helper`,
+    then `.mock.` will be set to `None`.
+    """
+
+    retVal = False
+    try:
+        # Does the package exist?
+        from vector.apps.DataAPI import mock_helper
+
+        # The version of mock helper this version of the extension supports
```

Comment:
Add a comment that we are hard-coding etc.
So over time we will might have to add our own translation layer here.
How can we stay in sync? 
It seems like we should have some assert that kills our testing if we ever see a version > our version

Reply:
> Add a comment that we are hard-coding etc.
> So over time we will might have to add our own translation layer here.

Done.

> How can we stay in sync?

Problem for a later date when the API actually has "breaking changes".

> It seems like we should have some assert that kills our testing if we ever see a version > our version

`vmockGenerator.py` will "immediately fail" if `vpythonHasCodedMockSupport()` returns `False` (which is this function).